### PR TITLE
[forge] create land-blocking test with 20 validators and 10 VFNs

### PR DESCRIPTION
**Pull request by bchocho**
### Description

Also tune the VFN configs and txn emitter to allow ~1K TPS to go through each VFN.

### Test Plan

Check the land blocking test. It consistently reaches ~7K TPS

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2587)

### CODE REVIEW

**Comment by sitalkedia** [comment]
The latency of transaction submission is still very high - https://cloud.us.humio.com/k8s/search?query=forge-pr-2587%20%20%7C%20%22k8s.container_name%22%3Dfullnode%0A%7C%20api%7C%20%20path%20%0A%7C%20data.path%20%3D%20%22%2Ftransactions%22%0A%7C%20unit%3Aconvert%28data.elapsed%2C%20to%3D%22ms%22%29%0A%7C%20timeChart%28function%3Davg%28data.elapsed%29%29&live=false&start=1659732587488&end=1659746489289

**Comment by sitalkedia** [comment]
@bchocho - Seems like we are running out of CPU on FNs - https://o11y.aptosdev.com/grafana/explore?left=%7B%22datasource%22:%22Remote%20Prometheus%20Devinfra%22,%22queries%22:%5B%7B%22expr%22:%22rate(container_cpu_usage_seconds_total%7Bcontainer%3D~%5C%22fullnode%5C%22,%20namespace%3D%5C%22forge-pr-2587%5C%22%7D%5B30s%5D)%22,%22format%22:%22time_series%22,%22interval%22:%22%22,%22intervalFactor%22:1,%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PCD0403638111AF12%22%7D%7D,%7B%22expr%22:%22rate(container_cpu_usage_seconds_total%7Bcontainer_name%3D~%5C%22fullnode%5C%22,%20namespace%3D%5C%22forge-pr-2587%5C%22%7D%5B30s%5D)%22,%22format%22:%22time_series%22,%22interval%22:%22%22,%22intervalFactor%22:1,%22refId%22:%22B%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22PCD0403638111AF12%22%7D%7D%5D,%22range%22:%7B%22from%22:%221659752827000%22,%22to%22:%221659754105000%22%7D%7D&orgId=1. I was under impression that we bumped up the cpu but its still using 3 cores - may be we need to apply that change? cc - @rustielin

**Comment by rustielin** [comment]
> I was under impression that we bumped up the cpu but its still using 3 cores

It was not applied earlier, but I just applied the change. Let's have it configurable through the Forge interfaces though

**Comment by bchocho** [comment]
Thanks @rustielin ! Let me give it another go

**Comment by bchocho** [comment]
Using the beefier FNs, we see > 6K TPS.

The latency of transaction submission looks reasonable now, actually much lower than when we run with no FNs https://cloud.us.humio.com/k8s/search?query=forge-pr-2587%20%20%7C%20%22k8s.container_name%22%3Dfullnode%0A%7C%20api%7C%20%20path%20%0A%7C%20data.path%20%3D%20%22%2Ftransactions%22%0A%7C%20unit%3Aconvert%28data.elapsed%2C%20to%3D%22ms%22%29%0A%7C%20timeChart%28function%3Davg%28data.elapsed%29%29&live=false&start=1659978473309&end=1659989826336

@sitalkedia what is the peak load txn emitter will try to send? Is it perhaps maxed out at ~1K?

If it's not the txn emitter, then it may be due to contention as the CPU maxes out at ~600%. If contention is the case, we may need to batch client submitted transactions.

But we should also think about what's a reasonable TPS supported on a single FN -- are we already there?

**Comment by sitalkedia** [comment]
@bchocho - Do you have the grafana dashboard link with beefier FN? Also, can we may be try bumping up the number of FNs to see if that helps?

**Comment by bchocho** [comment]
@sitalkedia this is the mempool dashboard for the 6.8K run: https://o11y.aptosdev.com/grafana/d/mempool-brian-updated/mempool?from=1660001738000&to=1660002904000&var-Datasource=Remote%20Prometheus%20Devinfra&var-namespace=forge-pr-2587&var-chain_name=forge-0&var-kubernetes_pod_name=All&orgId=1

I'm pretty sure we'll get > 7K with 7 FNs (instead of 6). Will kick that off soon

File: testsuite/run_forge.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ FORGE_RUNNER_MODE=${FORGE_RUNNER_MODE:-k8s}
 FORGE_NAMESPACE_KEEP=${FORGE_NAMESPACE_KEEP:-false}
 FORGE_NAMESPACE_REUSE=${FORGE_NAMESPACE_REUSE:-false}
 FORGE_ENABLE_HAPROXY=${FORGE_ENABLE_HAPROXY:-false}
-FORGE_TEST_SUITE=${FORGE_TEST_SUITE:-land_blocking}
+FORGE_TEST_SUITE=${FORGE_TEST_SUITE:-bench_with_fullnode}
```
Comment (sitalkedia) [inline testsuite/run_forge.sh:41]: @bchocho - Just learned that you need to change this to 

```
FORGE_TEST_SUITE=bench_with_fullnode

```
This is due to recent change where GH started passing this default value.

**Comment by sitalkedia** [comment]
We also need to understand why the FN is consuming so much of CPU just for transaction submission and querying the accounts - can we run perf on it and generate a flamegraph?

**Comment by sitalkedia** [comment]
@bchocho - I see the latest run did ~7k TPS - what magic was performed 😃 ?

**Comment by bchocho** [comment]
No magic just added one more FN (6 -> 7). We were at over 6K with 6. Maybe we can commit this while we investigate further? 

I do need to see if we can get this throughput with failover re-enabled.

**Comment by bchocho** [comment]
2K reads and 1K writes per FN is probably OK for now given overall system throughput?

**Comment by sitalkedia** [comment]
How do we feel about making the test with FNs as land blocking instead of only with validators?

**Comment by bchocho** [comment]
Sgtm I guess it's ~25% more expensive

**Comment by sitalkedia** [comment]
Here's what we can do - instead of running 30 validators, we can change the topology to run 20 validators and 10 FNs - that way we can put the FNs in the critical testing path and test functionalities like mempool and state sync as well. Can we run this test several times and see if its stable with 10 FNs and we can achieve consistently 7K TPS before making this land blocking?

**Comment by bchocho** [comment]
OK, let me clean this up and run it a few times. I will also push these configs into the fullnode configs (via yaml). We probably don't want them for the validators.

**Comment by bchocho** [comment]
With default_failovers = 3 (the previous default) we see reduced TPS
> performance benchmark : 6677 TPS, 4368 ms latency, 8250 ms p99 latency,no expired txns

Duplicate txns are at around 7K per validator. https://o11y.aptosdev.com/grafana/d/mempool-brian-updated-2/mempool-brian-updated-2?from=1660176634000&to=1660177604000&var-Datasource=Remote%20Prometheus%20Devinfra&var-namespace=forge-pr-2587&var-chain_name=forge-0&var-kubernetes_pod_name=All&orgId=1

I'm going to try failovers = 1

File: terraform/helm/aptos-node/values.yaml [hunk 2]
```diff
@@ -66,6 +66,9 @@ fullnode:
   affinity: {}
   config:
     max_inbound_connections: 100
+    shared_mempool_max_concurrent_inbound_syncs: 16
```
Comment (sitalkedia) [inline terraform/helm/aptos-node/values.yaml:69]: @bchocho - Are these configs that should be default for FNs instead of just being for `Forge`?

Comment (bchocho) [reply terraform/helm/aptos-node/values.yaml:69]: Yes, these yaml's will affect the defaults outside of Forge as well.

Comment (sitalkedia) [reply terraform/helm/aptos-node/values.yaml:69]: Seems like txns per account bumping up 5 -> 10 didn't help? Let's dial that down in that case?

**Review by sitalkedia — APPROVED** [review]

**Pull request merged**